bin/merge_jsons.py

```diff
@@ -5,41 +5,74 @@
 import json
 import argparse
 
-from collections import Counter
 
 def __main__():
-    parser = argparse.ArgumentParser(description="")
-    parser.add_argument('-s', "--single_input", help='Single input JSON report')
-    parser.add_argument('-i', "--input", help='Input directory with JSON reports')
-    parser.add_argument('-p', "--prefix", help="Prefix for output")
+    parser = argparse.ArgumentParser(
+        description="Merge multiple JSON reports into one."
+    )
+    parser.add_argument("-s", "--single_input", help="Single input JSON report")
+    parser.add_argument("-i", "--input", help="Input directory with JSON reports")
+    parser.add_argument("-p", "--prefix", help="Prefix for output")
 
     args = parser.parse_args()
 
     if len(sys.argv) <= 1:
         parser.print_help()
         sys.exit(1)
 
+    # Flatten irregular list function
+    def flatten(l):
+        """Flatten list using recursion."""
+        if isinstance(l, list):
+            for el in l:
+                if isinstance(el, list):
+                    yield from flatten(el)
+                else:
+                    yield el
+        else:
+            return l
+
+    # Combine_dicts function
+    def combine_dicts(a, b):
+        joined_list = list(a.items()) + list(b.items())
+        d = {}
+        for key, value in joined_list:
+            d.setdefault(key, []).append(value)
+        return d
+
     # read in json reports
-    data = []
+    data = dict()
     if args.single_input:
-        with open(args.single_input, 'r') as infile:
-            data.append(Counter(json.load(infile)))
+        with open(args.single_input, "r") as infile:
+            data = combine_dicts(data, json.load(infile))
     else:
         for file in os.listdir(args.input):
             if file.endswith(".json"):
                 with open(file, "r") as infile:
```

Comment:
I think we should change that to:

```suggestion
                with open(os.path.join(args.input, file), "r") as infile:
```

Reply:
yeah right it does not matter for the pipeline because dir is ".", but to make it general that would be good!